src/devices/cpu/arcompact/arcompact.h

```diff
@@ -105,795 +40,751 @@ class arcompact_device : public cpu_device
+	// 0x0c - LE - Less than or equal (signed)
+	bool condition_LE() { return ((status32_check_z()) || (status32_check_n() && !status32_check_v()) ||  (!status32_check_n() && status32_check_v())) ; }
+	// 0x0d - HI - Higher than (unsigned)
+	bool condition_HI() { return ((!status32_check_c()) && (!status32_check_z())); }
+	// 0x0e - LS - Lower than or same (unsigned)
+	bool condition_LS() { return (status32_check_c() || status32_check_z()); }
+	// 0x0f - PNZ - Positive Non Zero
+	bool condition_PNZ() { return ((!status32_check_n()) && (!status32_check_z())); }
+
+	void check_interrupts();
+
+	/************************************************************************************************************************************
+	*                                                                                                                                   *
+	* 32-bit opcode handlers                                                                                                            *
+	*                                                                                                                                   *
+	************************************************************************************************************************************/
+
+	uint32_t handleop32_B_cc_D_s21(uint32_t op);
+	uint32_t handleop32_B_D_s25(uint32_t op);
+	uint32_t handleop32_BL_cc_d_s21(uint32_t op);
+	uint32_t handleop32_BL_d_s25(uint32_t op);
+
+	int32_t get_01_01_01_address_offset(uint32_t op);
+	uint32_t BRxx_takejump(uint32_t address, uint8_t n, int size);
+	bool BRxx_condition(uint8_t condition, uint32_t b, uint32_t c);
+	uint32_t handleop32_BRxx_reg_reg(uint32_t op, uint8_t condition);
+	uint32_t handleop32_BRxx_reg_imm(uint32_t op, uint8_t condition);
+
+	uint32_t handleop32_BREQ_reg_reg(uint32_t op);
+	uint32_t handleop32_BRNE_reg_reg(uint32_t op);
+	uint32_t handleop32_BRLT_reg_reg(uint32_t op);
+	uint32_t handleop32_BRGE_reg_reg(uint32_t op);
+	uint32_t handleop32_BRLO_reg_reg(uint32_t op);
+	uint32_t handleop32_BRHS_reg_reg(uint32_t op);
+	uint32_t handleop32_BBIT0_reg_reg(uint32_t op);
+	uint32_t handleop32_BBIT1_reg_reg(uint32_t op);
+	uint32_t handleop32_BREQ_reg_imm(uint32_t op);
+	uint32_t handleop32_BRNE_reg_imm(uint32_t op);
+	uint32_t handleop32_BRLT_reg_imm(uint32_t op);
+	uint32_t handleop32_BRGE_reg_imm(uint32_t op);
+	uint32_t handleop32_BRLO_reg_imm(uint32_t op);
+	uint32_t handleop32_BRHS_reg_imm(uint32_t op);
+	uint32_t handleop32_BBIT0_reg_imm(uint32_t op);
+	uint32_t handleop32_BBIT1_reg_imm(uint32_t op);
+	uint32_t handleop32_LD_r_o(uint32_t op);
+	uint32_t handleop32_ST_r_o(uint32_t op);
+
+	typedef uint32_t (*ophandler32)(void*,uint32_t,uint32_t,uint8_t);
+	typedef void (*ophandler32_ff)(void*,uint32_t,uint32_t);
+	typedef void (*ophandler32_mul)(void*,uint32_t,uint32_t);
+	typedef uint32_t (*ophandler32_sop)(void*,uint32_t, uint8_t);
```

Comment:
If you really want to do this with function pointers like this, why are you obfuscating it and making it more error prone by making the first parameter `void *` when you could make `arcompact_device &` and avoid having to cast back and forth?  Also, please give the arguments names in the function type declarations, and use `bool` for things you only need to do a truth test on.

These would be a lot clearer as:
```c++
    using ophandler32 = uint32_t (*)(arcompact_device &obj, uint32_t src1, uint32_t src2, bool set_flags);
    using ophandler32_ff = void (*)(arcompact_device &obj, uint32_t src1, uint32_t src2);
    using ophandler32_mul = void (*)(arcompact_device &obj, uint32_t src1, uint32_t src2);
    using ophandler32_sop = uint32_t (*)(arcompact_device &obj, uint32_t src1, bool set_flags);
```

Reply:
done